processor/src/main/java/org/mapstruct/ap/internal/model/source/Mapping.java

```diff
@@ -88,10 +96,51 @@
         return mappings;
     }
 
+    public static Mapping artificialMapping(String source, String target) {
```

Comment:
I used source ".*" as replacement for mapping all properties from source and because I don't know mapstruct so well it was easier for me to read all properties from source object, construct mapping relation to current object (kinda if I would manually write mapping for every property) and just reuse default functionality of library to do the rest. 
But if you think that more right way would be do it in BeanMappingMethod, I can try that as well.

Reply:
Thanks, I can try to do that myself, but if I stuck with some part then would be nice to have someone that could help with that part.

Reply:
I removed '.*' from source part, so if using target="." it will try to map everything from source property (and if source is missing, it will just do default thing and maps everything from source to target). Also moved actual mapping generation to BeanMappingMethod.